src/viser/infra/_messages.py

```diff
@@ -18,6 +18,42 @@
     ClientId = Any
 
 
+def _prepare_for_deserialization(value: Any, annotation: Type) -> Any:
+    # If annotated as a float but we got an integer, cast to float. These
+    # are both `number` in Javascript.
+    if annotation is float:
+        return float(value)
+    elif annotation is int:
+        return int(value)
+    elif get_origin(annotation) is tuple:
+        out = []
+        args = get_args(annotation)
+        if ... in args:
+            if len(value) < len(args) - 1:
+                warnings.warn(f"[viser] {value} does not match annotation {annotation}")
+                return value
+            ellipsis_index = args.index(...)
+            num_ellipsis = len(value) - len(args) + 2
+            args = (
+                args[: (ellipsis_index - 1)]
+                + tuple(args[ellipsis_index - 1] for _ in range(num_ellipsis))
+                + args[ellipsis_index + 1 :]
```

Comment:
I'm not really following the logic for these changes, it looks like this is trying to handle cases like `tuple[T1, T2, ..., T3]`, which doesn't seem valid in Python?

Reply:
Sorry, I didn’t realise it was invalid